.github/workflows/build-and-deploy.yml

```diff
@@ -205,6 +205,16 @@ jobs:
         with:
           password: ${{ secrets.PYPI_TOKEN }}
 
+  deploy-rust:
+    runs-on: kuzu-self-hosted-linux-building
```

Comment:
Since this is a very simple job that does not require compilation, consider move it to a GitHub-hosted runner instead of using our own infrastructure.

Reply:
The choice was deliberate. It's running a full build of the crate before uploading it, which is particularly important since the [rust API only includes certain files when packaging](https://github.com/kuzudb/kuzu/blob/master/tools/rust_api/Cargo.toml#L13-L24) both because the `tools/rust_api/kuzu-src` symlink breaks the way it usually omits untracked git files, and because there's a maximum upload size and including stuff like the dataset directory would make the result too large. There's no verification that it can be built with the included files in CI, so having this last-minute check is important (and while it's . Having the check in CI instead would be slow, but would probably be a lot less so with sccache set up correctly.

Reply:
Oh, that I missed. I confused it with `linux-self-hosted-testing`. Since the results of the build aren't what's being published, it probably should just use the testing environment shouldn't it? I don't know what exactly the difference between the two environments are but I'm assuming it's something to do with the libraries it links against.

Reply:
Yes the building environment uses a distro with old `glibc` so the prebuilt binaries can have better compatibility. You can use the testing environment since the binaries are not going to be published.

Reply:
I have updated it to use `kuzu-self-hosted-testing`